### ISSUE TITLE
We. need. faster. builds.

### ISSUE THREAD

**Issue by danieldietrich** [javaslang/javaslang#1594]
Currently the Benchmarks are also tested when unit tests run (both, on CI Server and on a local machine).

When cherry picking for a release, many builds are involved (after each cherry-picked commit). For 2.0.4 it were 47 commits.

In 2.0.3 one `mvn clear test` ran 01:29 min on my machine. On current master it runs 07:40 min.

That. is. too. long.

Overall build time

* 47 commits * 01:29 min = < 1 hour
* 47 commits * 07:40 min = 6 hours

The next release has nearly 700 commits.

---

Goal: Create faster builds
Non-Goals: Remove tests completely

---

Suggested solution:

* Locally run only non-benchmark unit-tests (by default). By switch (e.g. Maven profile) the benchmark tests can also be run locally
* On the CI server the benchmark unit tests should run by default to ensure nothing broke.

**Comment by danieldietrich** [comment]
**Note:** Next week I meet Stefan Oehme (Gradle Gmbh) in Kiel to create a Gradle build for Javaslang.

**Comment by paplorinc** [comment]
+1 for a Gradle build, though that's not the source of slowdown.
Our build is not very complicated, should be doable in a few days :) (I've done more Maven to Gradle transitions that I'm proud to admit ...).

Also, if you cherrypick locally and commit the result only once, resulting in a single build.

**Comment by danieldietrich** [comment]
We will move to Gradle and optimize the build #1277

**Comment by paplorinc** [comment]
@danieldietrich, why do you think Gradle will make the tests run faster?

**Comment by danieldietrich** [comment]
@paplorinc The travis CI build time does not matter to me. It is the local build time that matters when I cherry-pick commits for a release.
When switching to Gradle I will add options to disable GWT tests and Benchmarks. I will run these once before pushing.

**Comment by paplorinc** [comment]
As mentioned previously, you can do that today also, by cherrypicking everything locally, and push only at the end.

I don't think Gradle will do anything differently here. 

If Travis supports it, we may try parallel tests, however :) (probably possible in Maven also)

**Comment by danieldietrich** [comment]
I think it can be configured with Maven Surefire Fork option
